std/wasi/snapshot_preview1.ts

```diff
@@ -1556,4 +1555,41 @@ export default class Context {
       }),
     };
   }
+
+  /**
+   * Attempt to begin execution of instance as a command by invoking its
+   * _start() export.
+   *
+   * If the instance does not contain a _start() export, or if the instance
+   * contains an _initialize export an error will be thrown.
+   *
+   * The instance must also have a WebAssembly.Memory export named "memory"
+   * which will be used as the address space, if it does not an error will be
+   * thrown.
+   */
+  start(instance: WebAssembly.Instance) {
+    const { _start, _initialize, memory } = instance.exports;
+
+    if (
+      typeof memory === undefined || !(memory instanceof WebAssembly.Memory)
+    ) {
+      throw new TypeError("WebAsembly.instance must provide a memory export");
+    }
+
+    this.memory = memory;
+
+    if (typeof _initialize !== "function" && _initialize !== undefined) {
+      throw new TypeError(
+        "WebAsembly.instance export _initialize must not be defined",
+      );
+    }
```

Comment:
Not sure but shouldn't it be something like this?
```js
  if (!_initialize) {
    throw new TypeError(
      "WebAsembly.instance export _initialize must not be defined",
    );
  }
```

Reply:
Yikes.

Not quite but it is indeed wrong, should just be checking if _initialize is a function and throw if it is (there are two types of entry points, _initialize, or _start).